pkg/scheduler/plugins/gang/gang.go

```diff
@@ -0,0 +1,437 @@
+		if len(gang.Children) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+/******************Pod\PodGroup Event END******************/
+
+/******************Schedule Related BEG******************/
+func (gang *Gang) GetGangWaitTime() time.Duration {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.WaitTime
+}
+
+func (gang *Gang) GetChildrenNum() int {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return len(gang.Children)
+}
+
+func (gang *Gang) GetGangMinNum() int {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.MinRequiredNumber
+}
+
+func (gang *Gang) GetGangTotalNum() int {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.TotalChildrenNum
+}
+
+func (gang *Gang) GetGangMode() string {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.Mode
+}
+
+func (gang *Gang) GetGangAssumedPods() int {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return len(gang.WaitingForBindChildren) + len(gang.BoundChildren)
+}
+
+func (gang *Gang) GetGangScheduleCycle() int {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.ScheduleCycle
+}
+
+func (gang *Gang) GetChildScheduleCycle(pod *v1.Pod) int {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	podId := GetNamespaceSplicingName(pod.Namespace, pod.Name)
+	return gang.ChildrenScheduleRoundMap[podId]
+}
+
+func (gang *Gang) GetCreateTime() time.Time {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.CreateTime
+}
+
+func (gang *Gang) GetGangGroup() []string {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.GangGroup
+}
+
+func (gang *Gang) IsGangResourceSatisfied() bool {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.ResourceSatisfied
+}
+
+func (gang *Gang) IsGangScheduleCycleValid() bool {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	return gang.ScheduleCycleValid
+}
+
+func (gang *Gang) SetChild(pod *v1.Pod) {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	podId := GetNamespaceSplicingName(pod.Namespace, pod.Name)
+	gang.Children[podId] = pod
+	klog.Infof("SetChild, gangName:%v, childNAme:%v", gang.Name, podId)
+}
+
+func (gang *Gang) SetScheduleCycleValid(valid bool) {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	gang.ScheduleCycleValid = valid
+	klog.Infof("SetScheduleCycleValid, gangName:%v, valid:%v", gang.Name, valid)
+}
+
+func (gang *Gang) SetChildScheduleCycle(pod *v1.Pod, childCycle int) {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	podId := GetNamespaceSplicingName(pod.Namespace, pod.Name)
+	gang.ChildrenScheduleRoundMap[podId] = childCycle
+}
+
+func (gang *Gang) TryUpdateScheduleCycle() {
+	gang.lock.Lock()
+	defer gang.lock.Unlock()
+
+	num := 0
+	for _, childScheduleCycle := range gang.ChildrenScheduleRoundMap {
+		if childScheduleCycle == gang.ScheduleCycle {
+			num++
+		}
+	}
+
+	if num == gang.TotalChildrenNum {
+		gang.ScheduleCycleValid = true
+		klog.Infof("SetScheduleCycleValid, gangName:%v, valid:%v", gang.Name, true)
```

Comment:
this log should be in the end of the "if", and add "ScheduleCycle"

Reply:
done